float32 distL
float32 distM
float32 distR
int32[2] pixL
int32[2] pixM
int32[2] pixR